cmd/pdcli/alerts/cmd.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	printAlerts(alerts)
+
+	return nil
+}
+
+// getIncidents returns a string slice consisting IDs of the first 10 incidents.
+func getIncidents(client *pagerduty.Client, isHigh bool) ([]string, error) {
+
+	var incidentIDs []string
+	var status []string
+	var teams []string
+	var users []string
+
+	var opts pagerduty.ListIncidentsOptions
+
+	var aerr pagerduty.APIError
+
+	// Get current user details
+	user, err := client.GetCurrentUser(pagerduty.GetCurrentUserOptions{})
+
+	if err != nil {
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+	}
+
+	// Check the assigned-to flag
+	switch args.assignment {
+
+	case "team":
+		// Fetch incidents belonging to a specific team
+		opts.TeamIDs = append(teams, constants.TeamID)
+
+	case "silentTest":
+		// Fetch incidents assigned to silent test
+		opts.UserIDs = append(teams, constants.SilentTest)
+
+	case "self":
+		// Fetch incidents only assigned to self
+		opts.UserIDs = append(users, user.ID)
+	}
+
+	// Fetch only triggered, acknowledged incidents (not resolved ones)
+	opts.Statuses = append(status, constants.StatusTriggered, constants.StatusAcknowledged)
+
+	// Let the number of incidents fetched
+	opts.Limit = constants.AlertsLimit
+
+	// Check if flag is true
+	if args.low {
+		opts.Urgencies = []string{"low"}
+	} else if args.high {
+		opts.Urgencies = []string{"high"}
+	}
+
+	eps, err := client.ListIncidentsWithContext(context.Background(), opts)
+
+	if err != nil {
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+	}
+
+	// Check if there are no incidents returned
+	if len(eps.Incidents) == 0 {
+		fmt.Println("Currently there are no alerts assigned to " + args.assignment)
+		os.Exit(1)
+	}
+
+	for _, incident := range eps.Incidents {
+		incidentIDs = append(incidentIDs, incident.Id)
+	}
+
+	return incidentIDs, nil
+}
+
+// getIncidentAlerts returns all the alerts belong to a particular incident.
+func getIncidentAlerts(client *pagerduty.Client, incidentID string) ([]alert, error) {
+
+	var alerts []alert
+
+	var opts pagerduty.ListIncidentAlertsOptions
+
+	eps, err := client.ListIncidentAlertsWithContext(context.Background(), incidentID, opts)
+
+	if err != nil {
+		var aerr pagerduty.APIError
+
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+
+	}
+
+	for _, p := range eps.Alerts {
+
+		tempAlertObj := alert{}
+
+		tempAlertObj.incidentID = incidentID
+		tempAlertObj.name = p.Summary
+		tempAlertObj.severity = p.Severity
+		tempAlertObj.status = p.Status
+		tempAlertObj.clusterID = fmt.Sprint(p.Body["details"].(map[string]interface{})["cluster_id"])
```

Comment:
`ClusterProvisioningDelay` alerts (like for example) P637NF6) don't have cluster_id set, how will this lookup work in this case?

Reply:
There'll be an interactive feature which will let you select an alert and look into its details. That'll help I suppose?
I guess, I'll have to figure it out while working on [OSD-8117](https://issues.redhat.com/browse/OSD-8117)